desktop/cmp/panel/PanelModel.ts

```diff
@@ -9,18 +9,89 @@ import {
     managed,
     ManagedRefreshContextModel,
     PersistenceProvider,
-    PrefProvider,
+    PersistOptions,
+    PrefProvider, RefreshContextModel,
     RefreshMode,
     RenderMode,
     XH
 } from '@xh/hoist/core';
 import '@xh/hoist/desktop/register';
-import {action, makeObservable, observable} from '@xh/hoist/mobx';
+import {action, makeObservable, observable, comparer} from '@xh/hoist/mobx';
 import {wait} from '@xh/hoist/promise';
 import {throwIf} from '@xh/hoist/utils/js';
 import {isNil} from 'lodash';
 import {createRef} from 'react';
-import {ModalSupportModel} from '../modalsupport/ModalSupportModel';
+import {ModalSupportModel, ModalSupportOptions} from '../modalsupport/';
+
+export interface PanelConfig {
+
+    /** Can panel be resized? */
+    resizable?: boolean;
+
+    /** Redraw panel as resize happens? */
+    resizeWhileDragging?: boolean;
+
+    /** Can panel be collapsed, showing only its header? */
+    collapsible?: boolean;
+
+    /** Default size (in px) of the panel. */
+    defaultSize?: number;
+
+    /** Minimum size (in px) to which the panel can be resized. */
+    minSize?: number;
+
+    /** Maximum size (in px) to which the panel can be resized. */
+    maxSize?: number;
+
+    /** Default collapsed state. */
+    defaultCollapsed?: boolean;
+
+    /**
+     * Side towards which the panel collapses or shrinks. This relates
+     * to the position within a parent vbox or hbox in which the panel should be placed.
+     */
+    side?: 'top'|'bottom'|'left'|'right';
```

Comment:
Wonder if we should expose this as a global `Side` type? We have the same options in TabSwitcher

Reply:
Done!